### PR TITLE
PLU-241: [name-your-pipe-2]: replace create flow functionalities

### PR DESCRIPTION
**Pull request by m0nggh**
## Problem
We have too many "name your pipe" pipes and it is hard for us to tell what the user is doing at one glance, so we should make the user name their pipe so that it is easier for them to track their own pipes as well.

## Details
This PR focuses on replacing the create flow functionalities in plumber and removing those unnecessary ones

Frontend:
- Replaced create pipe on the main page
- Removed the `editor/create` route and all possible ways to create pipe in apps page
  - Cannot create flow under add new app connection, under app flows, under flows page

Backend:
- Removed `connectionId` and  `appKey` from the `create-flow` graphQL mutation
- Add UUID check to `getFlow` graphQL query

## TODO
- [x] Update e2e since it is using `/editor/create`
Need to approve e2e [PR](https://github.com/opengovsg/plumber-e2e/pull/27) fix after merging this

## Tests
- [x] Cannot create pipe anywhere else other than clicking on the `create pipe` button in the flows page
- [x] Pipe creation still works with the modal

### CODE REVIEW

**Comment by m0nggh** [comment]
* **#683** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/683?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#682** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/682?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#681** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/681?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @m0nggh and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/src/graphql/queries/get-flow.ts [hunk 1]
```diff
@@ -1,6 +1,13 @@
 import type { QueryResolvers } from '../__generated__/types.generated'
 
+const UUID_REGEX = /^[a-z,0-9,-]{36,36}$/
+
 const getFlow: QueryResolvers['getFlow'] = async (_parent, params, context) => {
+  // To avoid the gibberish error code if a user keys in an invalid editor route e.g. editor/123
+  if (!UUID_REGEX.test(params.id)) {
+    throw new Error('The pipe id in your URL is of an invalid format.')
+  }
+
```
Comment (ogp-weeloong) [inline packages/backend/src/graphql/queries/get-flow.ts]: nit: I would avoid regex if possible, plus UUIDv4 is actually stricter than this regex
```suggestion
  if (!z.string().uuid().safeParse(params.id).success) {
    throw new Error('Please provide a valid pipe ID.')
  }
```

File: packages/frontend/src/pages/Flows/index.tsx [hunk 2]
```diff
@@ -124,85 +119,85 @@ export default function Flows(): React.ReactElement {
...
-            <ConditionalIconButton
-              type="submit"
-              size="lg"
-              component={CreateFlowLink}
-              icon={<AddIcon />}
-              data-test="create-flow-button"
-            >
-              Create Pipe
-            </ConditionalIconButton>
+            <GridItem area="title">
```
Comment (ogp-weeloong) [inline packages/frontend/src/pages/Flows/index.tsx:141]: comment: niceeeee

**Review by ogp-weeloong — APPROVED** [review]
lgtm

File: packages/backend/src/graphql/queries/get-flow.ts (hunk 1 above)
Comment (m0nggh) [reply packages/backend/src/graphql/queries/get-flow.ts]: ok thanks!

**Pull request merged**